examples/quick-start-tail-sampling/README.md

```diff
@@ -0,0 +1,123 @@
+
+## Scenario
+
+In this scenario, we have on [Go API](./simple-go-service/) that sends Trace data to one instance of [OTel Collector](https://opentelemetry.io/docs/collector/), which samples 50% of the traces and sends it to [Jaeger](https://www.jaegertracing.io/) data store.
+
+```mermaid
+---
+title: Initial observability architecture
+---
+flowchart LR
+    GoAPI["Go API"]
+    OTelCol["OTel Collector"]
+    DataStore[("Jaeger")]
+
+    GoAPI -- Send traces --> OTelCol
+    OTelCol -- Forward 50% of Traces --> DataStore
+``````
+
+The collector configuration is defined by a `traces/jaeger` pipeline with one receiver, two processors (one to send traces in batch and the other set up the probabilistic sampling) and one exporter to Jaeger:
+
+```yaml
+receivers:
+  otlp:
+    protocols:
+      grpc:
+      http:
+
+processors:
+  batch:
+    timeout: 100ms
+  probabilistic_sampler:
+    hash_seed: 22
+    sampling_percentage: 50.0 # 50%
+
+exporters:
+  otlp/jaeger:
+    endpoint: jaeger:4317 # Send traces to Jager
+    tls:
+      insecure: true
+
+service:
+  pipelines:
+    traces/jaeger:
+      receivers: [otlp]
+      processors: [batch, probabilistic_sampler]
+      exporters: [otlp/jaeger]
+
+```
+
+However, to run a Trace-based test on this scenario and validate if the API is ok, we need to have the testing trace collected by our OTelCollector, how can we do that?
+
+Since Tracetest always starts a trace by adding a `tracetest=true` key/value to [TraceState](https://opentelemetry.io/docs/specs/otel/trace/api/#tracestate), that is [propagated](https://opentelemetry.io/docs/instrumentation/js/propagation/), we can filter these traces on the collector and send them to Tracetest.
+
+To do it we need to set up Tracetest to use an OTLP Data Store on its provision ([here](./tracetest/tracetest.provision.yaml)) and also set up another pipeline with a [Tail Sampling](https://opentelemetry.io/docs/concepts/sampling/#tail-sampling) processor on OTel Collector, filtering by TraceState `tracetest=true` value.
+
+We can do that by adding the following configuration to OTel Collector (the entire collector config YAML is [here](./tracetest/collector.config.yaml)):
+```yaml
+#...
+
+processors:
+  #...
+  tail_sampling:
+    decision_wait: 10s
+    num_traces: 100
+    expected_new_traces_per_sec: 10
+    policies:
+      [
+        {
+          name: Accept only traces that started from tracetest,
+          type: trace_state,
+          trace_state: {
+            key: tracetest,
+            values: ["true"]
+          }
+        }
+      ]
+
+exporters:
+  #...
+  otlp/tracetest:
+    endpoint: tracetest:4317 # Send traces to Tracetest.
+    tls:
+      insecure: true
+
+service:
+  pipelines:
+    #...
+    traces/tracetest:
+      receivers: [otlp]
+      processors: [batch, tail_sampling]
+      exporters: [otlp/tracetest, logging]
```

Comment:
Would be nice to have a third configuration mixing both approaches, still doing probabilistic sampling, but making sure all tracetest traces go to a datastore (either jaeger or tracetest)

Reply:
No, this is why I'm using an OTel Collector config to do the tail sampling.